letsencrypt/proof_of_possession.py

```diff
@@ -61,6 +62,13 @@ def perform(self, achall):
             if cert_key == achall.hints.jwk:
                 return self._gen_response(achall, key)
 
+        if os.path.isdir(str(self.config.key_dir)):
```

Comment:
`str` shouldn't be necessary. To fix unit tests in `proof_of_possession_test`, add a value for `key_dir` to the mocked configuration created on line 31. This can be accomplished with something like `self.config = mock.MagicMock(key_dir="/nonexistent/key/dir")`